lib/gollum-lib/page.rb

```diff
@@ -10,18 +10,35 @@ class << self
       #
       # query     - The String path to match.
       # entry     - The BlobEntry to check against.
-      # global_match - If true, find a File matching path's filename, but not its directory (so anywhere in the repo)
-      def path_match(query, entry, global_match = false, hyphened_tags = false, case_insensitive = false)
+      def global_path_match(query, entry, hyphened_tags = false, case_insensitive = false)
         return false if "#{entry.name}".empty?
         return false unless valid_extension?(entry.name)
-        entry_name = valid_extension?(query) ? entry.name : strip_filename(entry.name)     
         match_path = ::File.join([
-          '/',
-          global_match ? nil : entry.dir,
-        entry_name
+          '/', 
+          entry.dir,
+        entry.name
```

Comment:
I'm sure there's a Pathname variant you could use here, but I'm not sure it would be more elegant. In terms of consistency, you might want to consider it (e.g., `Pathname.new('/').join([etc. ...])`).

Reply:
The working `Pathname` variant would be:

```ruby
        match_path = Pathname.new('/').join(*[
          entry.dir,
          entry.name
          ].compact
        ).to_s
```
But since it adds a splat and a `.to_s` call, I'm also not sure if it's more elegant. I'll leave it as is for now, if you think consistency is important here I'm happy to go for that.

Reply:
My mission is consistency in the use of Pathname, but this is your call.

Reply:
Went for the `Pathname` variant :)